src/query/interpret/eval.hpp

```diff
@@ -467,7 +469,105 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
         throw QueryRuntimeException("Invalid property name {} for Graph", prop_name);
       }
       default:
-        throw QueryRuntimeException("Only nodes, edges, maps and temporal types have properties to be looked-up.");
+        throw QueryRuntimeException(
+            "Only nodes, edges, maps, temporal types and graphs have properties to be looked up.");
+    }
+  }
+
+  TypedValue Visit(AllPropertiesLookup &all_properties_lookup) override {
+    TypedValue::TMap result(ctx_->memory);
+    std::map<std::string, TypedValue> result_;
+
+    auto expression_result = all_properties_lookup.expression_->Accept(*this);
+    switch (expression_result.type()) {
+      case TypedValue::Type::Null:
+        return TypedValue(ctx_->memory);
+      case TypedValue::Type::Vertex: {
+        const auto vertex = expression_result.ValueVertex();
+        auto properties = *vertex.Properties(view_);
+        for (const auto &[property_id, value] : properties) {
```

Comment:
```suggestion
        for (const auto vertex = expression_result.ValueVertex(); const auto &[property_id, value] : *vertex.Properties(view_);) {
```

It's picky and feel free to ignore it but I like it because of automatic scope